app/mailers/application_mailer.rb

```diff
@@ -42,4 +42,10 @@ def group_mail(users, options)
   def set_locale_for(user)
     FastGettext.set_locale(user.locale.blank? ? "en" : user.locale)
   end
+
+  def set_url
+    unless (@url = URI.parse(Setting[:foreman_url])).present?
+      raise ":foreman_url is not set, please configure in the Foreman Web UI (More -> Settings -> General)"
```

Comment:
@domcleal - proposed a fix for this issue [9613](http://projects.theforeman.org/issues/9613) also in this PR. Left this ```raise``` for backwards compatibility